pandas/core/series.py

```diff
@@ -4546,7 +4544,7 @@ def rename(
         if mapper is None:
             mapper = index
         if callable(mapper) or is_dict_like(mapper):
```

Comment:
can also revert these lines (and the 2 above, and the call involving `mapper` below)

Reply:
Yeah, I am familiar with the fact that I haven't reverted all the changes of `series.rename` just wanted to run it by you so that you can confirm whether I am on the right track?

Reply:
will force push the rest of the changes if you give me a go-ahead.

Reply:
yes, go ahead

Reply:
Pushed the rest of the changes, review them whenever you are free!!

Reply:
cool, thanks - last thing would be to remove the extra test from pandas/tests/series/methods/test_rename.py (the one which uses `mapper`)

Reply:
<img width="653" alt="changed" src="https://user-images.githubusercontent.com/75975268/144064137-39568674-6901-4be3-9eea-a385defa7484.png">
These were the changes that you did inside test_rename.py but now it looks like this -
<img width="609" alt="rename_py" src="https://user-images.githubusercontent.com/75975268/144064320-d4f17e5e-497c-4bff-8de6-449bc14142a4.png">

Reply:
The function `test_rename_no_method_no_index` no longer exists. So should I remove the function `test_rename_method_and_index` ?

Reply:
Yes you can remove `test_rename_method_and_index`

Reply:
The Entire function?

Reply:
yup, if we're removing `mapper` it's no longer relevant